GVRf/Framework/jni/engine/importer/importer_jni.cpp

```diff
@@ -53,8 +53,14 @@ Java_org_gearvrf_NativeImporter_readFileFromAssets(JNIEnv * env,
     char* buffer = (char*) malloc(sizeof(char) * size);
     AAsset_read(asset, buffer, size);
 
-    AssimpImporter* assimp_scene = Importer::readFileFromAssets(
-            buffer, size);
+    AssimpImporter* assimp_scene = NULL;
+    /* Retrieve the file extension and pass it as hint */
+    char* dot = strrchr(native_string, '.');
+    if (!dot || dot == native_string) {
+        assimp_scene = Importer::readFileFromAssets(buffer, size, 0);
+    } else {
+        assimp_scene = Importer::readFileFromAssets(buffer, size, dot + 1);
```

Comment:
Can we make this part reused, thinking of avoiding the similar codes written more than once? If yes,
1)	Make this part as a separate function, then make it called from NativeImporter_readFromByteArray(), too
2)	Change the arg jstring hint of NativeImporter_readFromByteArray() to jstring filename
3)	Change and simplify codes accordingly
4)	And, do the same for other callers in the future